src/collections/components/AddLabelModal/AddLabelModal.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { Grid } from '@mui/material';
+import { Modal } from '../../../_shared/components';
+import { AddLabelFormConnector } from '../';
+import { Label } from '../../../api/generatedTypes';
+
+interface AddLabelModalProps {
+  /**
+   * Whether the modal is visible on the screen or not.
+   */
+  isOpen: boolean;
+
+  /**
+   * Toggle the AddLabelModalProps to show/hide as necessary.
+   */
+  toggleModal: VoidFunction;
+
+  /**
+   * state variable from Parent component LabelList Page
+   * */
+  setLabelsList: React.Dispatch<React.SetStateAction<Label[]>>;
+}
```

Comment:
We could create a function that calls the state setter function instead of passing the state function directly to the child

```suggestion
  setLabelsList: (labelsList: Label[]) => void;
}
```

And then we can pass a function to the child component as a prop which calls the state setter e.g

```
// something like this
const setLabels = (labelList: Label[]) => {setLabelList(labeList);}
```
Apologies for misleading you!

Reply:
What about not manipulating the contents of the list directly? I know our cache management (specifically Apollo cache) within the app is a hot stinking mess, but we should at least make it a _consistent_ mess :). Then, once we've got some time set aside to deal with it, we could refactor it in a single sweep. 

@katerinachinnappan , what we'd usually do here is flush the cache for the `labels` query on the page by destructuring a helper function - `refetch` - from the original query and using it to flush the cache once the mutation is completed. It is supposed to work without this manual intervention but doesn't. [Example here](https://github.com/Pocket/curation-admin-tools/blob/9255721fb0cd1ba3d1ad1e5721defa2741085318/src/curated-corpus/pages/CorpusPage/CorpusPage.tsx#L25).

Reply:
Yeah even better^ lets use `refetch`, that way we get the label list from the query itself and maintain that one source of truth